Migrate internal user tests

Cruikshanks opened the pull request:
This change migrates the legacy internal user tests to this project. The legacy tests were

- create-user.spec.js
- reset-password.spec.js
- search-for-licence.spec.js
- view-contacts.spec.js

The tests we are migrating were all in a folder named `user/`. But as we're migrating them we are seeing that doesn't really make much sense. Most we're just placing in the root as they are standalone tests.

In the legacy reset-password test validation and the journey were mixed in. To make a clearer distinction about what we're testing we have split them into 2; journey and validation.

This is the same thing we did for the internal reset password migration.

andrewhick commented:
Maintenance step to ensure tests also pass in production with the data available.

Cruikshanks committed:
Migrate internal user tests

This change migrates the legacy internal user tests to this project. The legacy tests were

- create-user.spec.js
- reset-password.spec.js
- search-for-licence.spec.js
- view-contacts.spec.js

Cruikshanks committed:
Migrate internal reset password tests

In the legacy tests validation and the journey is mixed in. To make a clearer distinction about what we're testing we split the test into 2; journey and validation.

This is the same thing we did for the internal reset password migration.

Cruikshanks committed:
Move create user test

The tests we are migrating were all in a folder named `user/`. But as we're migrating them we are seeing that doesn't really make much sense.

So, left as is `create-user.cy.js` would be the only test in `user/`. To us, it makes more sense to move it to the root.

Cruikshanks committed:
Migrate search for licence

Cruikshanks committed:
Migrate view licence contacts

Cruikshanks committed:
Merge branch 'main' into migrate-internla-user-tests
